pkg/api/datasources.go

```diff
@@ -124,8 +127,24 @@ func DeleteDataSourceByName(c *models.ReqContext) Response {
 	return Success("Data source deleted")
 }
 
+func validateParams(params models.DataSourceParams) Response {
```

Comment:
Maybe a bit overkill for now. Why not just send in string argument?

Reply:
@marefr I also thought about about sending only the string, but then I figured we logically want to validate the parameters, and quite possibly more than just the URL, so we might as well prepare for it. I don't see any downside to let the validation function receive all parameters?

Reply:
Thinking it over a second time though, maybe it can be better solved through a shared embedded struct instead of an interface though?

Reply:
@marefr I discussed with @bergquist and he also thought it would be better to leave the embedded struct refactoring out of this PR, so I switched to a validation function only receiving the URL, as you suggested.